### DISCUSSION TITLE
Howto parse XML Data?

### DISCUSSION THREAD

**Discussion by klauweg** [micropython/micropython#14113] [General]
I try to parse xml data received by requests module.
I found xmltok. But it seems to expect a file-like object instead of a string.
The requests module has this stream option together with raw-results. But there goes something wrong with the encoding.
Do you have any Tips on what i should focus on?

**Comment by scruss** [comment]
I can't help with the stream vs string issue, but [ThunderEX/uxml2dict: xmltodict on micropython](https://github.com/ThunderEX/uxml2dict) seems to provide a more humane XML parser than xmltok.

Its example code is missing a test XML file, so I saved the first example from [<text> - SVG: Scalable Vector Graphics | MDN](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text#example) as `vector-text.svg`, and the example produced the following JSON:

```json
{"svg": {"@viewBox": "0 0 240 80", "@xmlns": "http://www.w3.org/2000/svg", "style": {"#text": ".small {\n      font: italic 13px sans-serif;\n    }\n    .heavy {\n      font: bold 30px sans-serif;\n    }\n\n    /* Note that the color of the text is set with the    *\n     * fill property, the color property is for HTML only */\n    .Rrrrr {\n      font: italic 40px serif;\n      fill: red;\n    }\n  "}, "text": [{"@x": "20", "@y": "35", "@class": "small", "#text": "My"}, {"@x": "40", "@y": "35", "@class": "heavy", "#text": "cat"}, {"@x": "55", "@y": "55", "@class": "small", "#text": "is"}, {"@x": "65", "@y": "55", "@class": "Rrrrr", "#text": "Grumpy!"}]}}
```

I could work with this structure more easily than writing a low-level parser with xmltok.

**Reply by klauweg** [reply]
As far as i understand, uxml2dict uses xmltok as well for tokenizing.

**Reply by scruss** [reply]
Yes, it does. But unlike using xmltok, uxml2dict creates a much higher level object.

Could you write your request to file and parse that with uxml2dict? Not ideal, I know

**Reply by klauweg** [reply]
Maybe this could be feasible. But this application is so simple that i now use a regex to get the Information out of the xml data.
